Simplified/Book/Models/NYPLBookRegistry.h

```diff
@@ -29,7 +29,15 @@ typedef NS_ENUM(NSInteger, NYPLBookState);
 
 @end
 
+#if FEATURE_AUDIOBOOKS
+// Hiding warnings for protocol defined in Swift is not accessible in ObjC
+#pragma clang diagnostic push
```

Comment:
@ettore From the previous PR ([link](https://github.com/NYPL-Simplified/Simplified-iOS/pull/1606#discussion_r978741090)), we decided to declare the conformance in the `NYPLBookRegistry.m` file. However, that would lead to compiling error when we dependency inject this class into the business logic, because the conformance cannot be seen on the Swift side.

Reply:
We will need to implement all the functions in this protocol, otherwise it will not compile since Swift compiler does not see the `NYPLBookRegistry` conforming to the protocol. Maybe we can implement the required functions and call the ObjC functions from the Swift extension?